src/test/java/DilemmaDetector/Simulator/ChangeLineApplierTest.java

```diff
@@ -0,0 +1,60 @@
+package DilemmaDetector.Simulator;
+
+import generator.BaseScenarioGenerator2;
+import generator.Model;
+import javafx.collections.SetChangeListener;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.semanticweb.owlapi.apibinding.OWLManager;
+import org.semanticweb.owlapi.model.OWLOntology;
+import org.semanticweb.owlapi.model.OWLOntologyCreationException;
+import org.semanticweb.owlapi.model.OWLOntologyManager;
+import project.MyFactory;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+
+public class ChangeLineApplierTest {
+    private static final double MOVING_TIME = 5.0;
+    //each TIME_PART we check if there is a collision between main vehicle and some different entity
+    private static final double TIME_PART = 0.01;
+
+    public static final String baseIRI = "http://webprotege.stanford.edu/";
+    private Model model;
+
+    @Before
+    public void init() throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, OWLOntologyCreationException {
+        OWLOntologyManager ontologyManager = OWLManager.createOWLOntologyManager();
+        OWLOntology ontology = ontologyManager.loadOntologyFromOntologyDocument(new File("src/main/resources/traffic_ontology.owl"));
+        MyFactory factory = new MyFactory(ontology);
+        BaseScenarioGenerator2 baseScenarioGenerator2 = new BaseScenarioGenerator2(factory, baseIRI);
+        this.model = baseScenarioGenerator2.generate();
```

Comment:
Chyba, żeby przetestować ChangeLineApplier, czyli coś do zmiany akcji, to potrzebujesz tylko rigid body, weather, numery linii i szerokość linii, nie potrzebujesz odnosić się do ontologii, tworzyć factory, ani nawet scenariusza czy modelu, zmockój te rzeczy, które potrzebujesz, albo poprzez narzędzie do mockowania, albo poprzez po prostu stworzenie obiektów tej klasy.